File: aoc2017/day5.swift
//
//  day5.swift
//  aoc2017
//
//  Created by Chris McElroy on 7/23/21.
//

import Foundation

func day5() {
	let max = input.count
	var ins = input.map { Int($0)!}
	
	var i = 0
	var s = 0
	while i.isin(0..<max) {
		let temp = ins[i]
		ins[i] += ins[i] >= 3 ? -1 : 1
		i += temp
		s += 1
	}
	print(s)
}

File: go/day5.go
package main

func day5() {
	jumps := inputInts()
	steps := 0
	i := 0
	for i >= 0 && i < len(jumps) {
		temp := jumps[i]
		jumps[i] += 1
		if temp >= 3 {
			jumps[i] -= 2
		}
		i += temp
		steps += 1
	}

	println(steps)
}
